rest/ProductService.js

```diff
@@ -82,11 +88,21 @@ class ProductService {
             throw new IllegalArgumentError(INVALID_QUANTITY_ERR);
         }
     }
+    
 
-    /**
-     * TODO
-     * Validate serial numbers
-     */
+    #assertProductIsActive(productId) {
+        const product = this.#productPersistence.getById(productId);
+        if (!product.isActive) {
+            throw new Error(INVALID_IS_ACTIVE_ERR);
```

Comment:
Replace all new Error() in validations with new IlegalArgumentError(), this way the server will return 400 and not 500